### PR TITLE
Suit Storage Unit QoL changes and tweaks [NEeds SpriTe]

### PR DESCRIPTION
**Pull request by warface1234455**
Suit storage unit changes
# Document the changes in your pull request

- **When hacked suit storage, it will be very devastating to mob**
<details><summary>hacked spoiler</summary>

- Extend the cycle time (3->7)
- Give mob a lot of burn damages (20, 36)
- Increase 500 rads per cycle (3500 rads total)
- Burn mob
- emits radiation
- Makes mob scream
- When done cycling, delete everything inside except for mob and spread bad smoke
- (Video is cropped to reduce the size)
### Not hacked

https://user-images.githubusercontent.com/89688125/199537152-4c3d15a2-cfe4-483d-83b1-de5500adeee6.mp4

### Hacked

https://user-images.githubusercontent.com/89688125/199536941-e4611cbf-6f58-4aa4-bab4-9708a33285b6.mp4

</details>

- Suit storage desc is changed to fit the current feature
- Add sound for suit storage unit

1. Close sound

https://user-images.githubusercontent.com/89688125/200490372-8f51fb54-df5d-4994-bfc7-3857076effcf.mp4

2. Open sound

https://user-images.githubusercontent.com/89688125/200490378-6e808a54-72dd-4919-9781-95e0c775106a.mp4

- Suit storage can now be buit
 
- Fixes being able to open suit storage unit with alt click while panel is open
 
- Fixes being able to open suit storage unit with alt click while machine is running
 

# Wiki Documentation
- Suit storage desc is changed to fit the current feature
- Add sound for suit storage unit
- Suit storage can now be buit
- Fixes being able to open suit storage unit with alt click while panel is open
- Fixes being able to open suit storage unit with alt click while machine is running
- **When hacked suit storage, it will be very devastating to mob**
<details><summary>emagged spoiler(Sprite is not updated)</summary>

- Extend the cycle time (3->7)
- Give mob a lot of burn damages (30, 46)
- Increase 500 rads per cycle (3500 rads total)
- emits radiation
- Burn mob
- shock mob
- Makes mob scream
- When done cycling, delete everything inside except for mob

</details>

# Changelog

:cl: Warface1234455
spellcheck: Suit storage desc is changed to fit the current feature
soundadd: Add sound for suit storage unit
rscadd: Suit storage can now be buit
bugfix: Fixed being able to open machine with alt click while panel is open for suit storage unit
bugfix: Fixed being able to open suit storage unit with alt click while machine is running
/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
Ok but can you still microwave people in them even if it doesn't clear the rads?

**Comment by ShoziOffical** [comment]
> Ok but can you still microwave people in them even if it doesn't clear the rads?

Yes

**Comment by Cartlord** [comment]
The sprite seems really unfitting.

**Comment by ShoziOffical** [comment]
> The sprite seems really unfitting.

you got better sprite then?

**Comment by VaelophisNyx** [comment]
I'm curious why we need this & why we need to make normal storage units NOT scrub rads?

**Comment by warface1234455** [comment]
> I'm curious why we need this & why we need to make normal storage units NOT scrub rads?

Read the pr smh

**Comment by ShoziOffical** [comment]
> I'm curious why we need this & why we need to make normal storage units NOT scrub rads?

I'm only removing the ability for the normal storage unit to clear mob radiation not the whole contamination clearing, the advanced one instead gets the ability to clear mob radiation

**Comment by JamieD1** [comment]
> > I'm curious why we need this & why we need to make normal storage units NOT scrub rads?
> 
> I'm only removing the ability for the normal storage unit to clear mob radiation not the whole contamination clearing, the advanced one instead gets the ability to clear mob radiation

But why? You are inventing the problem for your solution.

**Comment by ShoziOffical** [comment]
> But why? You are inventing the problem for your solution.

The normal suit storage unit tech is insufficient to do the mob radiation clearing, the advanced one is more advanced and well developed so it is capable of doing so @JamieD1

**Comment by ShoziOffical** [comment]
Cornflakes

**Comment by ShoziOffical** [comment]
CORNFLAKE AGAIN

**Comment by ShoziOffical** [comment]
Removed map changes

**Comment by JamieD1** [comment]
> > But why? You are inventing the problem for your solution.
> 
> The normal suit storage unit tech is insufficient to do the mob radiation clearing, the advanced one is more advanced and well developed so it is capable of doing so @JamieD1

Yeah but the normal one can scrub radiation right now, and the sprite does look out of place.

**Comment by ShoziOffical** [comment]
> > > But why? You are inventing the problem for your solution.
> > 
> > 
> > The normal suit storage unit tech is insufficient to do the mob radiation clearing, the advanced one is more advanced and well developed so it is capable of doing so @JamieD1
> 
> Yeah but the normal one can scrub radiation right now, and the sprite does look out of place.

Just an upgrade for it @JamieD1, the sprite does be fitting imo, Cuackless finds it the same

**Comment by cuackles** [comment]
The old 3/4 sprite looked out of place cus it was 3/4, the current one should fit better

**Comment by cuackles** [comment]
that's how the new one looks with the new perspective Wich makes it fit more, if you still have an issue with it the coloring could also be changed to a duller boring color so it pops less into view

**Comment by ShoziOffical** [comment]
@JamieD1 ^^^

**Comment by ShoziOffical** [comment]
welp Jamie said it's a different art style and still does look out of place, i will have to remove it now :(

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -21,14 +22,16 @@
 	state_open = FALSE
 	var/locked = FALSE
 	panel_open = FALSE
-	var/safeties = TRUE
+	var/safeties = FALSE
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: why are the safeties set to FALSE when it is first built?

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -431,8 +439,10 @@
 		if("door")
 			if(state_open)
 				close_machine()
+				playsound(src, 'sound/machines/oven/oven_close.ogg', 75, TRUE)
 			else
 				open_machine(0)
+				playsound(src, 'sound/machines/oven/oven_open.ogg', 75, TRUE)
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: Oven sounds really don't work here I feel

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -442,14 +452,20 @@
...
+			if(!helmet && !mask && !suit && !storage && !occupant)
 				return
-			else if(!helmet && !mask && !suit && !storage && !occupant)
-				return
-			else
-				if(occupant)
-					var/mob/living/mob_occupant = occupant
+			else 
+				if(uv_super)
+					say("ERROR: Decontamination process is going over safety limit!!")
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: This flavor text would probably work best as "ERROR: Decontamination Process Failure" or something "Decontamination is going over safety limit!!" feels... half baked? I guess

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -442,14 +452,20 @@
...
+					uv_cycles = 7
+				else
+					say("Please wait untill the decontamination process is completed.")
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: **Typo:** change `untill` to `until`

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -469,10 +485,18 @@
...
 		return
+	if(panel_open)
+		to_chat(user, span_warning("Close the panel first!"))
+		return
+	if(uv)
+		to_chat(user, span_warning("You cannot open the gate while the cycle is running!"))
+		return
 	if(state_open)
 		close_machine()
+		playsound(src, 'sound/machines/oven/oven_close.ogg', 75, TRUE)
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: Once again, oven sounds feel out of place here.

File: code/modules/research/designs/machine_designs.dm [hunk 6]
```diff
@@ -691,7 +691,15 @@
...
-/datum/design/board/crystallizer
+/datum/design/board/suit_storage_unit
+	name = "Machine Design (Suit Storage Unit)"
+	desc = "The circuit board for a suit storage unit."
+	id = "suit_storage_unit"
+	build_path = /obj/item/circuitboard/machine/suit_storage_unit
+	category = list ("Engineering Machinery")
+	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING
+
+/datum/design/board/decontamination
```
Comment (ReddicusDragon) [inline code/modules/research/designs/machine_designs.dm]: why is this renamed to `/datum/design/board/decontamination` from `/datum/design/board/crystallizer`?

File: code/game/machinery/suit_storage_unit.dm [hunk 7]
```diff
@@ -226,37 +223,43 @@
...
-				visible_message(span_warning("[src]'s door slides open, barraging you with the nauseating smell of charred flesh."))
+				visible_message(span_warning("[src]'s gate creaks open with a loud whining noise."))
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 50, TRUE)
+			QDEL_NULL(helmet)
+			QDEL_NULL(suit)
+			QDEL_NULL(mask)
+			QDEL_NULL(storage)
+			shock()
+		else
+			say("The decontamination process is completed, thank you for your patient.")
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: **Typo:** change `patient` to `patience`

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]
I dislike some of the design choices here. There is also some other general issues I've noticed, highlighted by comments.

**Comment by ReddicusDragon** [comment]
> > > But why? You are inventing the problem for your solution.
> > 
> > 
> > The normal suit storage unit tech is insufficient to do the mob radiation clearing, the advanced one is more advanced and well developed so it is capable of doing so @JamieD1
> 
> Yeah but the normal one can scrub radiation right now, and the sprite does look out of place.

This still doesn't scrub radiation from mobs, does it?

File: code/game/machinery/suit_storage_unit.dm (hunk 1 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: Accidentally left it when i was reverting other shits

File: code/game/machinery/suit_storage_unit.dm (hunk 2 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: It does

File: code/game/machinery/suit_storage_unit.dm (hunk 5 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: It does once again

File: code/game/machinery/suit_storage_unit.dm (hunk 4 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: Right

File: code/game/machinery/suit_storage_unit.dm (hunk 7 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: Yep

**Comment by ShoziOffical** [comment]
> This still doesn't scrub radiation from mobs, does it?

Nope, still does

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: How is it a failure when its frying someone, it clearly working at that moment but the strength of it is over limits

File: code/modules/research/designs/machine_designs.dm (hunk 6 above)
Comment (ShoziOffical) [reply code/modules/research/designs/machine_designs.dm]: Thanks for letting me know

**Comment by KillerOrcaCora** [comment]
I’m a bit confused here can someone help me out, we are removing the ability to scrub rads from the normal one? But there will be a advance one to cover up for it?

**Comment by ShoziOffical** [comment]
> I’m a bit confused here can someone help me out, we are removing the ability to scrub rads from the normal one? But there will be a advance one to cover up for it?

No, its still the same i removed the advanced one because the sprite is unfitting

File: code/game/machinery/suit_storage_unit.dm (hunk 2 above)
Comment (ReddicusDragon) [reply code/game/machinery/suit_storage_unit.dm]: It doesn't.

File: code/game/machinery/suit_storage_unit.dm [hunk 8]
```diff
@@ -1,11 +1,12 @@
 // SUIT STORAGE UNIT /////////////////
 /obj/machinery/suit_storage_unit
 	name = "suit storage unit"
-	desc = "An industrial unit made to hold and decontaminate irradiated equipment. It comes with a built-in UV cauterization mechanism. A small warning label advises that organic matter should not be placed into the unit."
+	desc = "An industrial unit made to hold and decontaminate irradiated equipment as well as organic. It comes with a built-in UV cauterization mechanism. A small warning label advises that UV bulb light must not be glowing."
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm:4]: I enjoy this description

**Comment by KillerOrcaCora** [comment]
DM me the png of the sprite Jamie and cuackles said was out of place, if you still want to go that route that is and I could make you something better

File: code/game/machinery/suit_storage_unit.dm [hunk 9]
```diff
@@ -226,37 +223,43 @@
...
-			mask = null
-			qdel(mask)
-			storage = null
-			qdel(storage)
-			// The wires get damaged too.
-			wires.cut_all()
-		else
-			if(!occupant)
-				visible_message(span_notice("[src]'s door slides open. The glowing yellow lights dim to a gentle green."))
+			say("ERROR: PLEASE CONTACT SUPPORT!!")
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: Recommend 1 exclamation instead of 2

File: code/game/machinery/suit_storage_unit.dm [hunk 10]
```diff
@@ -431,8 +439,10 @@
 		if("door")
 			if(state_open)
 				close_machine()
+				playsound(src, 'sound/machines/oven/oven_close.ogg', 75, TRUE)
```
Comment (ReddicusDragon) [inline code/game/machinery/suit_storage_unit.dm]: oven_close and oven_open simply do not work here, though oven sounds work for baking the organic

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (ReddicusDragon) [reply code/game/machinery/suit_storage_unit.dm]: "ERROR: Decontamination Process exceeds safety limit" feels more proper, for example.

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]

File: code/game/machinery/suit_storage_unit.dm (hunk 10 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: Sigh

File: code/game/machinery/suit_storage_unit.dm (hunk 8 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm:4]: Aigh

File: code/game/machinery/suit_storage_unit.dm (hunk 9 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: K

File: code/game/machinery/suit_storage_unit.dm (hunk 10 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: Guh ok ill find a replacement

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (ShoziOffical) [reply code/game/machinery/suit_storage_unit.dm]: I like this one

**Comment by ShoziOffical** [comment]
> DM me the png of the sprite Jamie and cuackles said was out of place, if you still want to go that route that is and I could make you something better

Drafted untill @KillerOrcaCora makes better sprite

**Comment by Bibby0110** [comment]
Can you reopen this when orca makes a better sprite, puts it further up the queue

**Comment by ShoziOffical** [comment]
ya

**Pull request closed without merging**